File: frontend/src/types/invitation.ts
export type ISODateString = string; // ISO-8601 timestamp from backend (UTC)

export interface Invitation {
  id: string;
  token: string;
  companyId: string;
  companyName: string;
  expiresAt: ISODateString;
  isUsed: boolean;
  usedAt: ISODateString | null;
  usedByUserId: string | null;
  usedByEmail: string | null;
}

export interface CreateInvitationRequest {
  companyId: string;
  expirationHours?: number;
}

export interface ValidateInvitationRequest {
  token: string;
}

File: backend/TypesettingMIS.Core/Entities/Invitation.cs
namespace TypesettingMIS.Core.Entities;

public class Invitation : BaseEntity
{
    public string Token { get; set; } = string.Empty;
    public Guid CompanyId { get; set; }
    public DateTimeOffset ExpiresAt { get; set; }
    public bool IsUsed { get; set; }
    public DateTimeOffset? UsedAt { get; set; }
    public Guid? UsedByUserId { get; set; }
    public string? UsedByEmail { get; set; }

    // Navigation property
    public Company Company { get; set; } = null!;
}
